tensorflow/compiler/xla/service/gpu/tests/mlir_gemm_test.cc

```diff
@@ -59,33 +82,39 @@ TEST_F(GemmTest, SimpleCase1) {
               ElementsAreArray<float>({11, 16, 19, 28}));
 }
 
+std::vector<float> diag_matrix(size_t dim, float diag_value) {
+  std::vector<float> r(dim * dim);
```

Comment:
You could already pre-initialize all values to 0:
```
std::vector<float> r(dim * dim, 0.0f)
```
And then run just a single loop
```
for (size_t i = 0; i < dim; ++i) {
  r[i * dim + i] = diag_value;
}
```

Reply:
That does seem cleaner.